yunwoo/baekjoon/그리디/1817-짐-챙기는-숌.js

```diff
@@ -0,0 +1,39 @@
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on('line', function (line) {
+  input.push(line);
+}).on('close', function () {
+  const [n, m] = input[0].split(' ').map(Number);
+  const booksWeight = n ? input[1].split(' ').map(Number) : 0;
+  let weight = 0;
+  let result = 1;
+
+  if (n) {
+    for (let i = 0; i < n; i++) {
+      weight += booksWeight[i];
+
+      if (weight + booksWeight[i + 1] > m) {
+        result++;
+        weight = 0;
+      }
+    }
+  } else {
+    result = 0;
+  }
+
+  console.log(result);
```

Comment:
```suggestion
  const [n, m] = input[0].split(" ").map(Number);
  const booksWeight = n ? input[1].split(" ").map(Number) : null;
  let weight = 0;
  let result = 1;

  if (!booksWeight) return console.log(0);
  for (let i = 0; i < n; i++) {
    weight += booksWeight[i];

    if (weight + booksWeight[i + 1] > m) {
      result++;
      weight = 0;
    }
  }

  console.log(result);
```
- 풀이잘봤습니다!
- booksWeight를 현재 삼항 연산자로 split할지 0으로 넣을지 분기처리 중인데 이때 0대신 null을 넣고, booksWeight가 null일 경우 return console.log(0)을 통해 처음 if...else 블록문을 제거할 수 있습니다.

Reply:
- 아하!! `console.log()`를 두번 쓰면 안되는 줄 알고..있었읍니다..😅 역시 훨씬 깔끔하군요!!! 감사합니다 👍
- 0 보다 null이 더 의미가 정확하네요!!🫡